samples/bluetooth/bap_unicast_client/src/stream_lc3.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_core.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/sys_clock.h>
+
+#include <lc3.h>
+#include <sys/errno.h>
+#include <math.h>
+
+#include "stream_lc3.h"
+#include "stream_tx.h"
+
+LOG_MODULE_REGISTER(lc3, LOG_LEVEL_INF);
+
+#define LC3_MAX_SAMPLE_RATE       48000U
+#define LC3_MAX_FRAME_DURATION_US 10000U
+#define LC3_MAX_NUM_SAMPLES       ((LC3_MAX_FRAME_DURATION_US * LC3_MAX_SAMPLE_RATE) / USEC_PER_SEC)
+/* codec does clipping above INT16_MAX - 3000 */
+#define AUDIO_VOLUME              (INT16_MAX - 3000)
+#define AUDIO_TONE_FREQUENCY_HZ   400
+
+static int16_t audio_buf[LC3_MAX_NUM_SAMPLES];
+/**
+ * Use the math lib to generate a sine-wave using 16 bit samples into a buffer.
+ *
+ * @param buf Destination buffer
+ * @param length_us Length of the buffer in microseconds
+ * @param frequency_hz frequency in Hz
+ * @param sample_rate_hz sample-rate in Hz.
+ */
+static void fill_audio_buf_sin(struct tx_stream *stream)
+{
+	const unsigned int num_samples =
+		(stream->lc3_tx.frame_duration_us * stream->lc3_tx.freq_hz) / USEC_PER_SEC;
+	const int sine_period_samples = stream->lc3_tx.freq_hz / AUDIO_TONE_FREQUENCY_HZ;
+	const float step = 2 * 3.1415f / sine_period_samples;
+
+	for (unsigned int i = 0; i < num_samples; i++) {
+		const float sample = sinf(i * step);
+
+		audio_buf[i] = (int16_t)(AUDIO_VOLUME * sample);
+	}
+}
+
+static int extract_lc3_config(struct tx_stream *stream)
+{
+	const struct bt_audio_codec_cfg *codec_cfg = stream->bap_stream->codec_cfg;
+	struct stream_lc3_tx *lc3_tx = &stream->lc3_tx;
+	int ret;
+
+	LOG_INF("Extracting LC3 configuration values");
+
+	ret = bt_audio_codec_cfg_get_freq(codec_cfg);
+	if (ret >= 0) {
+		ret = bt_audio_codec_cfg_freq_to_freq_hz(ret);
+
+		if (ret > 0) {
+			if (ret == 8000 || ret == 16000 || ret == 24000 || ret == 32000 ||
+			    ret == 48000) {
+				lc3_tx->freq_hz = (uint32_t)ret;
+			} else {
+				LOG_ERR("Unsupported frequency for LC3: %d", ret);
```

Comment:
In audio we usually refer to samplerate instead of frequency, OTOH bluetooth refers to sampling frequency.
```suggestion
				LOG_ERR("Unsupported sampling frequency for LC3: %d", ret);
```

Reply:
Sure - It was implied before, but we can write it more explicitly